packages/style-engine/class-wp-style-engine.php

```diff
@@ -413,12 +448,17 @@ public function get_block_supports_styles( $block_styles, $options ) {
 			foreach ( $definition_group_style as $style_definition ) {
 				$style_value = _wp_array_get( $block_styles, $style_definition['path'], null );
 
+				// If the style value contains a reference to another value in the tree.
+				if ( isset( $style_value['ref'] ) ) {
```

Comment:
The rise in new global styles features such as `ref` and quirky ones like `--wp--style--block-gap` makes me think we can extract parsing to a new class, where this stuff can take place.